definitions/src/lib.rs

```diff
@@ -8,6 +8,7 @@ pub const RISCV_PAGESIZE: usize = 1 << RISCV_PAGE_SHIFTS;
 pub const RISCV_GENERAL_REGISTER_NUMBER: usize = 32;
 // 4 MB
 pub const RISCV_MAX_MEMORY: usize = 4 << 20;
+pub const RISCV_MAX_MEMORY_V0: usize = 4 << 20;
```

Comment:
I was confused here:

* If we want to rename `RISCV_MAX_MEMORY` so everyone notices the breaking change, we shouldn't keep `RISCV_MAX_MEMORY`
* If we want to keep `RISCV_MAX_MEMORY` there is no need to introduce `RISCV_MAX_MEMORY_V0`

Reply:
We may increase the value of RISCV_MAX_MEMORY in the future, but for VERSION0 it can only be 4M.

Reply:
What's a use case for `RISCV_MAX_MEMORY` in the future? I would assume in the future we implement a VM that works on bigger memory size, in that sense we could support many different memory sizies(perhaps 4G will be a hard limit until we do a full audit of the assembly interpreter), maybe we don't need `RISCV_MAX_MEMORY`?

Reply:
I think it will act as an upper limit to ensure that the memory_size passed in is smaller than it when initializing the vm.

Reply:
Personally I feel it's better to put those upper limit at CKB side. Once we do support creating VMs with arbitrary memory size, we should just accept a `u32` parameter at VM side, we don't need any upper limit IMHO.

Reply:
agree